### PR TITLE
Update build tools, support lib + gradle version

### PR DESCRIPTION
**Pull request by snodnipper**
Bump:
 * ~~build tools from 25.0.1 to 25.0.3~~
 * support lib from 25.0.1 to 25.3.1
 * ~~gradle from 2.2.2 to 2.3.2~~

(addresses Android Studio warnings)

### CODE REVIEW

**Comment by snodnipper** [comment]
perhaps this is not possible given things like: https://stackoverflow.com/questions/42895689/travis-ci-gets-stuck-on-starting-build-tools-in-android

File: gradle.properties [hunk 1]
```diff
@@ -14,3 +14,27 @@ POM_LICENCE_DIST=repo
...
 POM_DEVELOPER_NAME=Square, Inc.
+
+## Project-wide Gradle settings.
+#
+# For more details on how to configure your build environment visit
+# http://www.gradle.org/docs/current/userguide/build_environment.html
+#
+# The Gradle daemon aims to improve the startup and execution time of Gradle.
+# When set to true the Gradle daemon is to run the build.
+org.gradle.daemon=true
```
Comment (JakeWharton) [inline gradle.properties]: The daemon is on by default in 3.0+ so this is redundant to specify. Plus i'm not sure how this affects the `--no-daemon` flag which we want to use on CI

File: gradle.properties [hunk 2]
```diff
@@ -14,3 +14,27 @@ POM_LICENCE_DIST=repo
...
+#
+# Specifies the JVM arguments used for the daemon process.
+# The setting is particularly useful for tweaking memory settings.
+# Default value: -Xmx10248m -XX:MaxPermSize=256m
+org.gradle.jvmargs=-Xmx2048m -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
```
Comment (JakeWharton) [inline gradle.properties]: perm gen is gone in JDK 8 so there's no need to specify its size

File: gradle.properties [hunk 3]
```diff
@@ -14,3 +14,27 @@ POM_LICENCE_DIST=repo
...
+#
+# When configured, Gradle will run in incubating parallel mode.
+# This option should only be used with decoupled projects. More details, visit
+# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
+org.gradle.parallel=true
+#
+# Enables new incubating mode that makes Gradle selective when configuring projects.
+# Only relevant projects are configured which results in faster builds for large multi-projects.
+# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:configuration_on_demand
+org.gradle.configureondemand=true
```
Comment (JakeWharton) [inline gradle.properties]: This project is so small that these two options don't have any measurable effect. They only serve to clutter the output and make it non-deterministic on CI.

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,6 +1,6 @@
-#Wed Nov 26 14:54:50 EST 2014
+#Tue May 30 10:54:19 BST 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.14.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
```
Comment (JakeWharton) [inline gradle/wrapper/gradle-wrapper.properties]: This was updated to 3.5 on `master` already

**Comment by snodnipper** [comment]
Thanks Jake.  The build seems to hang on `:sample:lint`- the gradle.properties changes were just to see they could influence the hanging.  Not sure if you had any thoughts on this?

```console
:sample:assembleRelease
:sample:assemble
:sample:lint
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

**Comment by JakeWharton** [comment]
Nope. Saw that in another PR and reverted the version bump until 3.0
becomes stable.

**Comment by snodnipper** [comment]
updating only the support lib seems to build ok

**Pull request merged**